sql/core/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -377,8 +379,9 @@ final UTF8String getUTF8String(int rowId) {
       if (stringResult.isSet == 0) {
         return null;
       } else {
-        return UTF8String.fromAddress(null,
-          stringResult.buffer.memoryAddress() + stringResult.start,
+        mb.setAddressAndSize(stringResult.buffer.memoryAddress(), stringResult.buffer.capacity());
```

Comment:
why use `stringResult.buffer.capacity()`? can we do `mb.setAddressAndSize(stringResult.buffer.memoryAddress() + stringResult.start, stringResult.end - stringResult.start)`?

Reply:
Good catch, done